### PR TITLE
Update the 'possibilities for extensions' section of step-6.

### PR DESCRIPTION
**Pull request by bangerth**
This uses the code example provided by Konstantin Ladutenko.

### CODE REVIEW

**Comment by kostyfisik** [comment]
This code looks messy:
```c_cpp
if (center.distance (cell->center()) < 1e-5)
```
I suppose to rename
```c_cpp
const Point<dim> center;
```
both in degenerative and non-degenerative parts to 
```c_cpp
const Point<dim> grid_center;
```
so it reads much easier for me
```c_cpp
if (grid_center.distance (cell->center()) < 1e-5)
```
It is a good idea to add after the last sentence 
"You can find much more on this in
 step-49, step-53, and step-54, among other tutorial programs that cover
 the issue."

 a link to "Manifold description for triangulations" http://dealii.org/8.2.1/doxygen/deal.II/group__manifold.html section, which was very useful for me during coding, however, it took me a while to find this page in manual.

**Comment by bangerth** [comment]
Good points. I've incorporated them.

File: examples/step-6/doc/results.dox [hunk 1]
```diff
@@ -258,15 +265,222 @@ lead to an increase in the number of iterations as the number $N$ of
 degrees of freedom grows, for example ${\cal O}(N^\alpha)$; this, in
 turn, leads to a total growth in effort as ${\cal O}(N^{1+\alpha})$
 since each iteration takes ${\cal O}(N)$ work. This behavior is
-undesirable, we would really like to solve linear systems with $N$
+undesirable: we would really like to solve linear systems with $N$
 unknowns in a total of ${\cal O}(N)$ work; there is a class
-of preconditioners that can achieve this, namely geometric (@ref
-step_16 "step-16") or algebraic (step-31) multigrid
+of preconditioners that can achieve this, namely geometric (step-16)
```
Comment (kronbichler) [inline examples/step-6/doc/results.dox]: geometric multigrid is also shown in step-37 and step-39. It's good to mention it here (or say that are more examples than just step-16.

File: examples/step-6/doc/results.dox [hunk 2]
```diff
@@ -258,15 +265,222 @@ lead to an increase in the number of iterations as the number $N$ of
...
+
+          static const SphericalManifold<dim> boundary;
+          triangulation.set_all_manifold_ids_on_boundary(0);
+          triangulation.set_manifold (0, boundary);
+
+          const Point<dim> mesh_center;
+          for (typename Triangulation<dim>::active_cell_iterator cell = triangulation.begin_active();
+               cell != triangulation.end(); ++cell)
+            if (mesh_center.distance (cell->center()) > cell->diameter()/10)
+              cell->set_all_manifold_ids (0);
```
Comment (Rombur) [inline examples/step-6/doc/results.dox]: Any reason why you use cell->diameter()/10? Doesn't this condition always true?

**Comment by kronbichler** [comment]
The new mesh looks indeed much better. Please fix the two remaining issues and merge yourself.

File: examples/step-6/doc/results.dox (hunk 2 above)
Comment (kostyfisik) [inline examples/step-6/doc/results.dox]: It is just to skip the central cell. Any small positive number should be fine.

**Pull request merged**